capitals.rb

```diff
@@ -150,3 +150,66 @@
     name: "Wyoming",
     capital: "Cheyenne"
 }]
+
+
+# shuffle states during each turn
+states = states.shuffle
+
+# states to include a key to keep track of number of times they get it right and the number of times they get it wrong
+game_states = states.each do |state|
+  state[:name] = state[:name]
+  state[:capital] = state[:capital]
+  state[:correct] = 0
+  state[:wrong] = 0
+  state[:times_displayed] = 0
+end
+
+# puts welcome message
+puts "Welcome to 'Let's Learn our State Capitals!' We'll give you a state, then you'll type what you think the capital is. Type 'play' to begin!"
+
+while true
+  start = gets.chomp
+  break if start == "play" || start == "Play"
+end
+
+# loop through all 50 states
+while true
+  states.each do |state|
+    puts "What's the capital of #{state[:name]}? Type your answer or 'hint' to get a hint."
+    # prompt user to name the capital of the state
+    answer = gets.chomp
+    # logic for if user wants a hint
+    if answer == "hint"
+      # regular expression to print first three letters
+      first_letters = state[:capital].scan(/\b(...)/i).join
```

Comment:
this is dope, but i think there is simple ruby string methods that can do this same sort of thing.